services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt

```diff
@@ -1746,7 +1747,7 @@ class HakemusServiceITest(
         }
 
         @Test
-        fun `works with kaivuilmoitus`() {
+        fun `kaivuilmoitus with existing johtoselvitys works`() {
```

Comment:
The name doesn't follow the naming convention we have been following for a while now. The name I gave this test wasn't very good either.

Reply:
I split the tests into multiple nested tests as you suggested below and renamed them also.